src/main/java/com/lgtm/easymoney/controllers/SearchController.java

```diff
@@ -25,8 +25,13 @@ public SearchController(SearchService searchService) {
 
   @GetMapping("/id/{id}")
   @Operation(summary = "Method to retrieve the public profile of a user by user ID.")
-  public ResponseEntity<SearchRsp> searchById(@PathVariable(value = "id") String id) {
-    return searchService.searchById(Long.valueOf(id));
+  public ResponseEntity<SearchRsp> searchById(@PathVariable(value = "id") String id) throws Exception {
+    if (!id.isEmpty()){
+      Long uid = Long.parseLong(id);
+      return searchService.searchById(uid);
+    }else{
+      throw new Exception("User ID can't be empty.");
```

Comment:
use our own ex handler?

Reply:
You can just use @PathVariable(...) Long id, and use @NotNull there as well. See https://stackoverflow.com/questions/23325413/what-runtimeexception-can-spring-crudrepository-throw